lib/galaxy/security/validate_user_input.py

```diff
@@ -97,11 +99,11 @@ def validate_email(trans, email, user=None, check_dup=True, allow_empty=False, v
     return message
 
 
-def validate_domain_resolves(domain):
+def validate_email_domain_name(domain: str) -> LiteralString:
     message = ""
     try:
-        socket.gethostbyname(domain)
-    except socket.gaierror:
+        dns.resolver.resolve(domain, "MX")
```

Comment:
According to the spec when there is no MX record the address resolution is supposed to fall back on the A record.

> If an empty list of MXs is returned,
> the address is treated as if it was associated with an implicit MX
> RR, with a preference of 0, pointing to that host. 

source: https://www.rfc-editor.org/rfc/rfc5321#section-5.1

Reply:
Maybe dnspython is smart enough to deal with such a situation? We'd need a test case.

Reply:
FQDNs. MX first and overriding, CNAME second and only with lack of MX, then queue, else error -- no other fall backs?

Reply:
I've randomly found a test case, a domain that has an A record but not MX: example.it

@martenson I've added a fall back to the A record that works for the test case.

@blankenberg That's a bit too cryptic for me, sorry! Feel free to suggest code changes or push to my branch!